integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/AppendDataVisitor.java

```diff
@@ -23,7 +24,8 @@ public AppendDataVisitor(OpenLineageContext context) {
   @Override
   public List<OpenLineage.OutputDataset> apply(LogicalPlan x) {
     // Needs to cast to logical plan despite IntelliJ claiming otherwise.
-    return PlanUtils.applyFirst(
-        context.getOutputDatasetQueryPlanVisitors(), (LogicalPlan) ((AppendData) x).table());
+    return new ArrayList<>(
+        PlanUtils.applyAll(
+            context.getOutputDatasetQueryPlanVisitors(), (LogicalPlan) ((AppendData) x).table()));
```

Comment:
Need to keep in mind this won't work soon once we rewrite `DatasourceV2RelationVisitor` into dataset builder.
But yeah, we can leave it for now.